src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1803,12 +1803,12 @@ bool LibraryCallKit::inline_vector_test() {
   if (opd1 == NULL || opd2 == NULL) {
     return false; // operand unboxing failed
   }
-  if (!Matcher::vectortest_need_second_argument(booltest == BoolTest::overflow, opd1->bottom_type()->isa_vectmask())) {
+  if (!Matcher::vectortest_needs_second_argument(booltest == BoolTest::overflow, opd1->bottom_type()->isa_vectmask())) {
     opd2 = opd1;
   }
```

Comment:
How about floating up this under line-1801, like:
```
Node* opd2 = NULL;
if (!Matcher::vectortest_needs_second_argument(...)) {
  opd2 = opd1;
} else {
  opd2 = unbox_vector(argument(5), ...);
}
if (opd1 == NULL || opd2 == NULL) {
  return false;
}
```
This can avoid the unnecessary vector unbox to `argument(5)` if the platform doesn't need opd2 during codegen. Thanks!